src/clab_vm_startup/conn_mode/traffic_control.py

```diff
@@ -0,0 +1,71 @@
+"""
+       Copyright 2021 Inmanta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+from pathlib import Path
+from textwrap import dedent
+from typing import List, Tuple
+
+from clab_vm_startup.host.host import Host
+from clab_vm_startup.host.nic import NetworkInterfaceController
+
+from .connection_mode import Connection, ConnectionMode
+
+
+class TrafficControlConnection(Connection):
+
+    TC_TAP_IFUP_PATH = Path("/etc/tc-tap-ifup")
+
+    def __init__(self) -> None:
+        super().__init__(ConnectionMode.TC)
+
+    def setup_host(self, host: Host) -> None:
+        """
+        For this connection mode, we need to deploy a script to setup the traffic control rules.
+        """
+        ifup_script = """
+            #!/bin/bash
+
+            TAP_IF=$1
```

Comment:
Where does this come from?

Reply:
Here: https://github.com/hellt/vrnetlab/blob/master/common/vrnetlab.py#L328

Reply:
We should attribute this correctly. If it is the same in the orignal repo, link to that.

Reply:
Done in https://github.com/inmanta/containerlab-vms/pull/1/commits/6d4269f01d4b064a7bcbe67326233a49a87b1a1f